### ISSUE TITLE
Slate is NOT uploading settings_data.json (since I edited it with CMS)

### ISSUE THREAD

**Issue by lastobelus** [Shopify/slate#1084]
### Problem

Since editing a section in the theme I am developing, Slate no longer uploads `config/settings_data.json` to my dev theme, either when running `yarn start`, or when editing it when `yarn watch` is active. (It's possible it never worked, even before using the theme section editor, but I'm not sure)

### Replication steps

1. edit & lint `config/settings_data.json`
2. run `yarn start`
3. slate claims to upload the correct count of files that would include `config/settings_data.json`, but it is unchanged in the theme editor.

### More Information

* tested with `cli.promptSettings` set to both true and false
* by responding both y & N to the prompt, I can see that when I tell it to go ahead and upload settings, the file count is one higher.
* no error is reported, all files are reported as uploaded.
* I've verified that the changes are present in the built version.
* I've verified the file **CAN** upload by uploading it directly with `theme deploy config/settings_data.json -d dist`, which works as expected.

**Comment by curiouscrusher** [comment]
Your most likely running into an issue with incompatible schema in Shopify. Have you confirmed that your `settings_data.json` is valid by copy/pasting it into your theme files with the Admin GUI editor? I've run into cases where after making Section changes in the admin and then trying to upload the data file it will upload it but not save due to incompatible schema based on the section settings. Slate does not warn you about this, although ThemeKit I believe does in the same scenario.

**Comment by lastobelus** [comment]
yes, I have confirmed that the settings_data.json is valid (and as I mentioned, I am able to upload  it directly with themekit from the build directory)

**Comment by jerrypena1** [comment]
Could it be possible you forgot to modify your .env file? there is a default
SLATE_IGNORE_FILES=config/settings_data.json

Unfortunately slate looks like its uploading ignore files although it will not.
